### PR TITLE
feat: support exlude windows

### PR DESCRIPTION
**Pull request by amythos**
add exlude_windows,   some windows do not need manage the window state for example splashscreen

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -219,4 +228,15 @@ impl Builder {
...
+  pub fn set_exlude_windows(mut self, exlude_windows: &[&str]) -> Self {
+    if !exlude_windows.is_empty() {
+      let mut exlude_set: HashSet<String> = HashSet::with_capacity(exlude_windows.len());
+      for win in exlude_windows {
+        exlude_set.insert(win.to_string());
+      }
+      self.exlude_windows = Some(exlude_set);
+    }
+    self
+  }
```
Comment (amrbashir) [inline src/lib.rs]: could you move this and `set_auto_show` to be above the `build` method?

Comment (amrbashir) [reply src/lib.rs]: and also rename them to `with_*` instead of `set_*`

File: src/lib.rs [hunk 2]
```diff
@@ -219,4 +228,15 @@ impl Builder {
 
     self
   }
+
```
Comment (amrbashir) [inline src/lib.rs]:
```suggestion

  /// Sets a blacklist of windows to ignore from tracking/restoring its state like splash screen windows
```

File: src/lib.rs [hunk 3]
```diff
@@ -219,4 +228,15 @@ impl Builder {
...
+  pub fn set_exlude_windows(mut self, exlude_windows: &[&str]) -> Self {
```
Comment (amrbashir) [inline src/lib.rs]:
```suggestion
  pub fn with_blacklist(mut self, blacklist: &[&str]) -> Self {
```

File: src/lib.rs [hunk 4]
```diff
@@ -124,11 +124,15 @@ impl<R: Runtime> WindowExt for Window<R> {
...
   auto_show: bool,
+  exlude_windows: Option<HashSet<String>>,
 }
 
 impl Default for Builder {
   fn default() -> Self {
-    Builder { auto_show: true }
+    Builder {
+      auto_show: true,
+      exlude_windows: None,
```
Comment (amrbashir) [inline src/lib.rs]:
```suggestion
      blacklist: None,
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (amythos) [reply src/lib.rs]: @amrbashir  i have fix that

File: src/lib.rs [hunk 5]
```diff
@@ -124,15 +124,35 @@ impl<R: Runtime> WindowExt for Window<R> {
...
+    self
+  }
+
+  pub fn with_blacklist(mut self, blacklist: &[&str]) -> Self {
+    if !blacklist.is_empty() {
+      let mut exlude_set: HashSet<String> = HashSet::with_capacity(blacklist.len());
+      for win in blacklist {
+        exlude_set.insert(win.to_string());
+      }
+      self.blacklist = Some(exlude_set);
```
Comment (amrbashir) [inline src/lib.rs]:
```suggestion
      let mut blacklist_set: HashSet<String> = HashSet::with_capacity(blacklist.len());
      for win in blacklist {
        exlude_set.insert(win.to_string());
      }
      self.blacklist = Some(blacklist_set);
```

File: src/lib.rs [hunk 6]
```diff
@@ -124,15 +124,35 @@ impl<R: Runtime> WindowExt for Window<R> {
...
-    Builder { auto_show: true }
+    Builder {
+      auto_show: true,
+      blacklist: None,
+    }
   }
 }
 
 impl Builder {
+  pub fn with_auto_show(mut self, auto_show: bool) -> Self {
```
Comment (amrbashir) [inline src/lib.rs:144]:
```suggestion
  /// Whether to enable or disable automatically showing the window
  /// 
  /// - `true`: the window will be automatically shown if the last stored state for visibility was `true`
  /// - `false`: the window will not be automatically shown by this plugin
  pub fn with_auto_show(mut self, auto_show: bool) -> Self {
```

File: src/lib.rs [hunk 7]
```diff
@@ -124,15 +124,35 @@ impl<R: Runtime> WindowExt for Window<R> {
...
+    self.auto_show = auto_show;
+    self
+  }
+
+  pub fn with_blacklist(mut self, blacklist: &[&str]) -> Self {
```
Comment (amrbashir) [inline src/lib.rs:151]:
```suggestion
  /// Sets a black list of windows that shouldn't be tracked and managed by this plugin
  /// for example splash screen widnows.
  pub fn with_blacklist(mut self, blacklist: &[&str]) -> Self {
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Review by amrbashir — APPROVED** [review]
Thank you

**Comment by amrbashir** [comment]
For future contributions, please sign you commits, thanks again.

**Comment by amythos** [comment]
> For future contributions, please sign you commits, thanks again.

thanks, i will sign my commits next time:)

**Pull request merged**